File: archive/test_api.py
#!/usr/bin/env python3
"""Simple test script to verify Google Gemini API key."""

import os
from google import genai
from dotenv import load_dotenv

def test_api_key():
    # Load .env file
    load_dotenv()
    api_key = os.getenv('GOOGLE_API_KEY')
    
    if not api_key:
        print("❌ GOOGLE_API_KEY environment variable not set")
        return False
    
    print(f"✅ API Key found: {api_key[:10]}...{api_key[-4:]}")
    
    try:
        # Initialize client
        client = genai.Client(api_key=api_key)
        
        # Simple test call
        response = client.models.generate_content(
            model="gemini-2.5-flash",
            contents=["Say hello"]
        )
        
        if response and response.text:
            print("✅ API Key is valid!")
            print(f"Response: {response.text}")
            return True
        else:
            print("❌ No response from API")
            return False
            
    except Exception as e:
        print(f"❌ API Error: {e}")
        return False

if __name__ == "__main__":
    test_api_key()

File: archive/test_api.R
#!/usr/bin/env Rscript

# Test script to debug API calls

library(ellmer)
library(tidyverse)

# Configuration  
source("config.R")

# Test the API connection
cat("Testing Gemini API connection...\n")

# Create chat
chat <- chat_google_gemini(
  model = EXTRACTION_MODEL,
  system_prompt = "You are a helpful assistant. Always respond with valid JSON."
)

# Simple test
test_prompt <- 'Please return a simple JSON array with one test object:
[
  {
    "test": "hello",
    "number": 123
  }
]'

cat("Making test API call...\n")

tryCatch({
  response <- chat$chat(test_prompt, echo = FALSE)
  cat("Response received:\n")
  cat("Length:", nchar(response), "\n")
  cat("Content:\n")
  cat(response)
  cat("\n\n")
  
  # Try to extract JSON from code blocks
  clean_response <- response
  if (grepl("```json", response)) {
    # Split by lines and find json boundaries
    lines <- strsplit(response, "\n")[[1]]
    start_idx <- which(grepl("```json", lines)) + 1
    end_idx <- which(grepl("^```$", lines))
    
    if (length(start_idx) > 0 && length(end_idx) > 0) {
      json_lines <- lines[start_idx:(end_idx - 1)]
      clean_response <- paste(json_lines, collapse = "\n")
      cat("Extracted JSON from code block\n")
    }
  }
  
  cat("Clean response:\n")
  cat(clean_response)
  cat("\n\n")
  
  # Try to parse as JSON
  json_data <- jsonlite::fromJSON(clean_response)
  cat("JSON parsing successful!\n")
  print(json_data)
  
}, error = function(e) {
  cat("Error:", e$message, "\n")
})

cat("\nTest complete.\n")